src/software/simulated_tests/simulated_test_fixture.py

```diff
@@ -141,9 +141,14 @@ def runner(
 
         time_elapsed_s = 0
 
+        # initialize the primitive block time to 0
+        primitive_block_time = 0
+
         eventually_validation_failure_msg = "Test Timed Out"
 
         while time_elapsed_s < test_timeout_s:
+            # get time before we execute the loop
+            primitive_block_time = time.time()
```

Comment:
nit: Could we rename the variables to be a bit more descriptive, since it's not just the primitive block time anymore.

Could also use two variables. e.g. `processing_start_time` -> `processing_time`